src/IssueViz.Security.UnitTests/Taint/TaintIssueToIssueVisualizationConverterTests.cs

```diff
@@ -130,262 +138,144 @@ public void Convert_FromSonarQubeIssue_IssueVizConverterCalledWithCorrectParamet
-    // {
-    //     var taintIssue = CreateTaintServerIssue(
-    //         mainLocation: CreateTaintServerLocation(
-    //             textRange: CreateTaintTextRange(1, 2, 3, 4, null)));
-    //
-    //     var expectedConvertedIssueViz = CreateIssueViz();
-    //
-    //     var issueVizConverter = new Mock<IAnalysisIssueVisualizationConverter>();
-    //     issueVizConverter
-    //         .Setup(x => x.Convert(It.IsAny<IAnalysisIssueBase>(), null))
-    //         .Returns(expectedConvertedIssueViz);
-    //
-    //     var testSubject = CreateTestSubject(issueVizConverter.Object);
-    //     var result = testSubject.Convert(taintIssue);
-    //
-    //     result.Should().Be(expectedConvertedIssueViz);
-    //     result.IsSuppressed.Should().BeFalse();
-    // }
-
-    private static TaintIssueToIssueVisualizationConverter CreateTestSubject(IAnalysisIssueVisualizationConverter issueVizConverter = null)
+    [DataTestMethod]
+    [DataRow(IssueSeverity.INFO, AnalysisIssueSeverity.Info)]
+    [DataRow(IssueSeverity.MINOR, AnalysisIssueSeverity.Minor)]
+    [DataRow(IssueSeverity.CRITICAL, AnalysisIssueSeverity.Critical)]
+    [DataRow(IssueSeverity.MAJOR, AnalysisIssueSeverity.Major)]
+    [DataRow(IssueSeverity.BLOCKER, AnalysisIssueSeverity.Blocker)]
+    public void StandardSeverity_Converts(IssueSeverity slCoreSeverity, AnalysisIssueSeverity expectedSeverity)
+    {
+        var taintVulnerabilityDto = CreateDefaultTaintDto(new StandardModeDetails(slCoreSeverity, default));
+
+        testSubject.Convert(taintVulnerabilityDto, "C:\\root");
+
+        issueVizConverter.Received()
+            .Convert(Arg.Is<ITaintIssue>(x => x.HighestSoftwareQualitySeverity == null && x.Severity == expectedSeverity));
+    }
+
+    [TestMethod]
+    public void StandardSeverity_InvalidSeverity_Converts()
+    {
+        var taintVulnerabilityDto = CreateDefaultTaintDto(new StandardModeDetails((IssueSeverity)999, default));
+
+        var act = () => testSubject.Convert(taintVulnerabilityDto, "C:\\root");
+
+        act.Should().Throw<ArgumentException>();
+        issueVizConverter.DidNotReceiveWithAnyArgs().Convert(default, default);
+    }
+
+    [TestMethod]
+    public void MqrSeverity_EmptySeverities_Throws()
+    {
+        var taintVulnerabilityDto = CreateDefaultTaintDto(new MQRModeDetails(default, []));
+
+        var act = () => testSubject.Convert(taintVulnerabilityDto, "C:\\root");
+
+        act.Should().Throw<ArgumentException>();
+        issueVizConverter.DidNotReceiveWithAnyArgs().Convert(default, default);
+    }
+
+    [DataTestMethod]
+    [DataRow(ImpactSeverity.BLOCKER, SoftwareQualitySeverity.Blocker)]
+    [DataRow(ImpactSeverity.HIGH, SoftwareQualitySeverity.High)]
+    [DataRow(ImpactSeverity.MEDIUM, SoftwareQualitySeverity.Medium)]
+    [DataRow(ImpactSeverity.LOW, SoftwareQualitySeverity.Low)]
+    [DataRow(ImpactSeverity.INFO, SoftwareQualitySeverity.Info)]
+    public void MqrSeverity_SingleSeverity_ReturnsIt(
```

Comment:
Maybe rename test to 
_SingleSeverity => _Single**SoftwareQualitySeverity** OR _Single**Impact**

Reply:
MqrSeverity is already in the title